examples/upgrade_forseti_with_v5.0/tutorial.md

```diff
@@ -0,0 +1,218 @@
+module, now version 5.0.0.  The steps outlined in this guide should not be needed
+when upgrading Forseti to versions newer than 2.23.
+
+If you have any
+questions about this process, please contact us by
+[email](mailto:discuss@forsetisecurity.org) or on
+[Slack](https://forsetisecurity.slack.com/join/shared_invite/enQtNDIyMzg4Nzg1NjcxLTM1NTUzZmM2ODVmNzE5MWEwYzAwNjUxMjVkZjhmYWZiOGZjMjY3ZjllNDlkYjk1OGU4MTVhZGM4NzgyZjZhNTE).
+
+## Prerequisites
+
+Before you begin the migration process, you will need:
+
+- A Forseti deployment of at least v2.18.0; follow the
+  [upgrade guide]({% link _docs/latest/setup/upgrade.md %}) as
+  necessary deployed via the [terraform-google-forseti Terraform module](https://github.com/forseti-security/terraform-google-forseti)
+- A version of the
+  [Terraform command-line interface](https://www.terraform.io/downloads.html)
+  in the 0.12 series.
+- The ID of the GCP project in which Forseti is deployed.
+- A service account in the organization with the
+  [roles required by the Terraform module](https://registry.terraform.io/modules/terraform-google-modules/forseti/google/4.3.0#iam-roles).
+- A
+  [JSON key file](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#creating_service_account_keys)
+  for the service account.
+
+## Configuring Terraform
+
+Terraform can assume the identity of a service account through a
+strategy called
+[Application Default Credentials](https://cloud.google.com/docs/authentication/production#providing_credentials_to_your_application)
+when provisioning resources. To enable this approach, set the
+appropriate environment variable to the path of the service account JSON
+key file:
+
+```sh
+export GOOGLE_APPLICATION_CREDENTIALS="PATH_TO_JSON_KEY_FILE"
+```
+## Backup the Terraform State
+In a shell, navigate to the folder containing your user-defined Terraform module, most likely in a **main.tf**.
+
+The following command will simultaneously backup your existing Terraform state and remove resource from the current state file.  This will not affect your existing Forseti deployment.
+```sh
+terraform state rm $(terraform state list)
+```
+## Update main.tf
+In order to support this upgrade, we'll need to update a few input variables.
+
+### Source
+The **source** will need to point to the root terraform-google-forseti module.
+```
+source = "terraform-google-modules/forseti/google"
+```
+If you have cloned the module to your local file system, you may set the **source** path to
+the directory containing the module.
+
+### Version
+The **version** will need to be 5.0.0.
+```
+version = "5.0.0"
+```
+If you have set the **source** path to the directory containing the module, omit the **version** variable.
+
+### Region
+If you have a **region**, it will need to be split into the cloudsql_region, server_region, and client_region variables.
+
+Before (example):
+```
+region = "us-central1"
+```
+After (example):
+```
+cloudsql_region = "us-central1"
```

Comment:
QQ: out of curiosity, if the cloudsql_region is changed, is the database still preserved?

Reply:
ack